cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -98,35 +106,75 @@ func (sub CosmosSub) Start(completionEvent *sync.WaitGroup) {
 	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
 	defer close(quit)
 
+	var lastProcessedBlock int64
+
+	data, err := sub.DB.Get([]byte(cosmosLevelDBKey), nil)
+	if err != nil {
+		log.Println("Error getting the last cosmos block from level db", err)
+		lastProcessedBlock = 0
+	} else {
+		lastProcessedBlock = new(big.Int).SetBytes(data).Int64()
+	}
+
 	for {
 		select {
-		case result := <-out:
-			tx, ok := result.Data.(tmTypes.EventDataTx)
+		case <-quit:
+			log.Println("we receive the quit signal and exit")
+			return
+
+		case e := <-results:
+			data, ok := e.Data.(tmTypes.EventDataNewBlock)
 			if !ok {
-				sub.SugaredLogger.Errorw("sifchain client failed to extract event data from new tx.",
-					errorMessageKey, err.Error())
+				sub.SugaredLogger.Errorw("sifchain client failed to extract event data from new block.",
+					"EventDataNewBlock", fmt.Sprintf("%v", e.Data))
 			}
+			blockHeight := data.Block.Height
 
+			// Just start from current block number if never process any block before
+			if lastProcessedBlock == 0 {
+				lastProcessedBlock = blockHeight
+			}
 			sub.SugaredLogger.Infow("new transaction witnessed in sifchain client.")
 
-			// Iterate over each event in the transaction
-			for _, event := range tx.Result.Events {
-				claimType := getOracleClaimType(event.GetType())
+			startBlockHeight := lastProcessedBlock + 1
+			fmt.Printf("cosmos process events from block %d to %d\n", startBlockHeight, blockHeight)
```

Comment:
Let's change this to a sugared logger call instead of a printf for consistency with `startingBlockHeight` and `currentBlockHeight` as keys.

Reply:
should use sugared logger, will fix it.